utils/network-helpers/src/plain_connection_tokio.rs

```diff
@@ -43,6 +50,12 @@ impl PlainConnection {
                     Ok(_) => {
                         if let Ok(x) = decoder.next_frame() {
                             sender_incoming.send(x.into()).await.unwrap();
+                        } else {
+                            error!("Error parsing incoming frame");
+                            if strict {
```

Comment:
I expect `decoder.next_frame` to return an Error at least half of the time that we call it. Cause:
1. decoder internal buffer is void and writable() return 6 
2. decoder internal buffer is filled with 6 bytes (the sv2 frame)
3. decoder deocde the frame and get payload len
4. writable() return the payload len
5. we fill the internal buffer with payload len
6. next_frame() return Ok(internal buffer)

For that I don't see where we could use the strict option

Reply:
Updated and fixed/changed to be more specific to noise handshake